### ISSUE TITLE
Should CA1825 warn when multitargeting?

### ISSUE THREAD

**Issue by JohanLarsson** [dotnet/roslyn-analyzers#4655]
```cs
var xs = new int[0];
```

When `<TargetFrameworks>net45;netcoreapp3.1</TargetFrameworks>`

Suppressing it via pragma brings up IDE0079

The way I found to satisfy both is on the clumsy side:

```cs
#pragma warning disable IDE0079 // Remove unnecessary suppression
#pragma warning disable CA1825 // Avoid zero-length array allocations
            var xs = new int[0];
#pragma warning restore CA1825 // Avoid zero-length array allocations
#pragma warning restore IDE0079 // Remove unnecessary suppression
```

```
dotnet_diagnostic.CA1825.severity = none
```

Is probably cleanest.

**Comment by Youssef1313** [comment]
Yes, I think it should warn when multitargeting.

The correct fix IMO is:

```csharp
var xs =
#if NET45
    new int[0];
#else
    Array.Empty<int>();
#endif
```
